integration-cli/docker_api_containers_test.go

```diff
@@ -715,6 +714,84 @@ func (s *DockerSuite) TestContainerApiCreateWithHostName(c *check.C) {
 	}
 }
 
+func (s *DockerSuite) TestContainerApiCreateWithDomainName(c *check.C) {
+	var domainName = "test-domain"
+	config := map[string]interface{}{
+		"Image":      "busybox",
+		"Domainname": domainName,
+	}
+
+	_, b, err := sockRequest("POST", "/containers/create", config)
+	if err != nil && !strings.Contains(err.Error(), "200 OK: 201") {
+		c.Fatal(err)
+	}
+
+	var container types.ContainerCreateResponse
+	if err := json.Unmarshal(b, &container); err != nil {
+		c.Fatal(err)
+	}
+
+	var id = container.ID
+
+	_, bodyGet, err := sockRequest("GET", "/containers/"+id+"/json", nil)
+
+	type configLocal struct {
+		Domainname string
+	}
+	type getResponse struct {
```

Comment:
@rajdeepd @duglin means you should use that type here, I've seen you used it above but this still needs to be changed! Thanks for you patience